testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java

```diff
@@ -541,4 +541,32 @@ public void testCastTimestampAsDecimal()
         onHive().executeQuery("DROP VIEW cast_timestamp_as_decimal_view");
         onHive().executeQuery("DROP TABLE cast_timestamp_as_decimal");
     }
+
+    @Test(groups = HIVE_VIEWS)
+    @Flaky(issue = RETRYABLE_FAILURES_ISSUES, match = RETRYABLE_FAILURES_MATCH)
+    public void testUnionBetweenCharAndVarchar()
```

Comment:
should this go to AbstractTestHiveViews?

Reply:
Probably not.

```
testing/bin/ptl test run --environment multinode --config config-default --  -t io.trino.tests.product.hive.TestHiveViewsLegacy.testUnionBetweenCharAndVarchar
```


```
Caused by: io.trino.spi.TrinoException: line 1:15: View 'hive.default.union_char_varchar_view' is stale or in invalid state: column [col] of type char(1024) projected from query view at position 0 cannot be coerced to column [col] of type varchar stored in view definition
tests               | 	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:48)
tests               | 	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:43)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.lambda$createScopeForView$56(StatementAnalyzer.java:2493)
tests               | 	at java.base/java.util.Optional.ifPresent(Optional.java:178)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.createScopeForView(StatementAnalyzer.java:2493)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.createScopeForView(StatementAnalyzer.java:2446)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:2238)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:493)
tests               | 	at io.trino.sql.tree.Table.accept(Table.java:60)
tests               | 	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:512)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.analyzeFrom(StatementAnalyzer.java:4560)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:3020)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:493)
tests               | 	at io.trino.sql.tree.QuerySpecification.accept(QuerySpecification.java:155)
tests               | 	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:512)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:520)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:1521)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:493)
tests               | 	at io.trino.sql.tree.Query.accept(Query.java:107)
tests               | 	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:512)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer.analyze(StatementAnalyzer.java:472)
tests               | 	at io.trino.sql.analyzer.StatementAnalyzer.analyze(StatementAnalyzer.java:461)
tests               | 	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:97)
tests               | 	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:86)
tests               | 	at io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:274)
tests               | 	at io.trino.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:209)
tests               | 	at io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:850)
tests               | 	at io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:153)
tests               | 	at io.trino.$gen.Trino_426_726_g33f1bc6_dirty____20231026_151108_2.call(Unknown Source)
```